plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -740,4 +740,13 @@ private void updateAllDependents(String parent, Plugin dependent, Map<String,Ver
             }
         }
     }
+
+    private Map<String, VersionNumber> difference(Map<String, VersionNumber> base, Map<String, VersionNumber> toAdd) {
```

Comment:
I find this method a bit confusing, most probably because I lack context. 

By reading the name I would expect it to calculate the difference between the maps passed, but by reading the code I see it adds the elements in the second one to the first one if not already present in it. But I'm not completely sure this is what we want because we are both modifying the first one and returning it.

Is 'adding if not present' the desired behaviour?